ts/components/services/ServiceMetadata.tsx

```diff
@@ -0,0 +1,157 @@
+import React from "react";
+import { View, StyleSheet } from "react-native";
+
+import I18n from "../../i18n";
+import { TranslationKeys } from "../../../locales/locales";
+import { H4 } from "../core/typography/H4";
+import TouchableDefaultOpacity from "../TouchableDefaultOpacity";
+import { capitalize } from "../../utils/strings";
+import customVariables from "../../theme/variables";
+import { ItemAction } from "../../utils/url";
+
+import ItemSeparatorComponent from "../ItemSeparatorComponent";
+
+import SectionHeader from "./SectionHeader";
+import LinkRow from "./LinkRow";
+
+type Props = {
```

Comment:
Since we have already auto-generated the ServicesMetadata type, could it be an improvement use it instead of its props? 
This avoid to pass a lot of props to this component and increase readability
Could it be a valid suggestion?

### here the working refactoring if you want to test it
```typescript
import React from "react";
import { View, StyleSheet } from "react-native";

import { OrganizationFiscalCode } from "italia-ts-commons/lib/strings";
import I18n from "../../i18n";
import { TranslationKeys } from "../../../locales/locales";
import { H4 } from "../core/typography/H4";
import TouchableDefaultOpacity from "../TouchableDefaultOpacity";
import { capitalize } from "../../utils/strings";
import customVariables from "../../theme/variables";
import { ItemAction } from "../../utils/url";

import ItemSeparatorComponent from "../ItemSeparatorComponent";

import { Service } from "../../../definitions/content/Service";
import { ServiceId } from "../../../definitions/backend/ServiceId";
import SectionHeader from "./SectionHeader";
import LinkRow from "./LinkRow";

type Props = {
  serviceId: ServiceId;
  organizationFiscalCode?: OrganizationFiscalCode;
  servicesMetadata: Service;
  getItemOnPress: (value: string, valueType?: ItemAction) => () => void;
};

const styles = StyleSheet.create({
  valueHeader: {
    marginTop: customVariables.spacerLargeHeight
  },
  valueRow: {
    flexDirection: "column"
  },
  valueItem: {
    flexDirection: "row",
    flex: 1,
    justifyContent: "space-between",
    marginVertical: 24
  },
  itemLabel: {
    flex: 1
  },
  itemValue: {
    flex: 1,
    marginRight: 8
  }
});

/**
 * Renders a dedicated section with a service's metadata and the header.
 */
const ServiceMetadata: React.FC<Props> = ({
  getItemOnPress,
  organizationFiscalCode,
  servicesMetadata,
  serviceId
}: Props) => (
  <>
    <SectionHeader iconName="io-phone" title={"services.contactsAndInfo"} />

    {/* links */}
    {servicesMetadata.web_url && (
      <LinkRow text={"services.visitWebsite"} href={servicesMetadata.web_url} />
    )}
    {servicesMetadata.support_url && (
      <LinkRow
        text={"services.askForAssistance"}
        href={servicesMetadata.support_url}
      />
    )}
    {servicesMetadata.app_ios && (
      <LinkRow text={"services.otherAppIos"} href={servicesMetadata.app_ios} />
    )}
    {servicesMetadata.app_android && (
      <LinkRow
        text={"services.otherAppAndroid"}
        href={servicesMetadata.app_android}
      />
    )}

    {/* touchable rows */}
    {organizationFiscalCode &&
      renderInformationRow(
        organizationFiscalCode,
        "profile.fiscalCode.fiscalCode",
        getItemOnPress(organizationFiscalCode, "COPY")
      )}
    {servicesMetadata.address &&
      renderInformationRow(
        servicesMetadata.address,
        "services.contactAddress",
        getItemOnPress(servicesMetadata.address, "MAP")
      )}
    {servicesMetadata.phone &&
      renderInformationRow(
        servicesMetadata.phone,
        "global.media.phone",
        getItemOnPress(`tel:${servicesMetadata.phone}`)
      )}
    {servicesMetadata.email &&
      renderInformationRow(
        servicesMetadata.email,
        "global.media.email",
        getItemOnPress(`mailto:${servicesMetadata.email}`)
      )}
    {servicesMetadata.pec &&
      renderInformationRow(
        servicesMetadata.pec,
        "global.media.pec",
        getItemOnPress(`mailto:${servicesMetadata.pec}`)
      )}
    {renderInformationRow(
      serviceId,
      "global.id",
      getItemOnPress(serviceId, "COPY")
    )}
  </>
);

const renderInformationRow = (
  value: string,
  label: TranslationKeys,
  onPress: () => void
) => (
  <View style={styles.valueRow}>
    <TouchableDefaultOpacity
      onPress={onPress}
      style={styles.valueItem}
      accessibilityRole={"button"}
    >
      <H4
        style={styles.itemValue}
        ellipsizeMode={"tail"}
        numberOfLines={1}
        color={"bluegrey"}
        weight={"Regular"}
      >
        {capitalize(I18n.t(label))}
      </H4>
      <H4
        style={styles.itemValue}
        ellipsizeMode={"tail"}
        numberOfLines={1}
        color={"blue"}
        weight={"SemiBold"}
      >
        {value}
      </H4>
    </TouchableDefaultOpacity>
    <ItemSeparatorComponent noPadded />
  </View>
);

export default ServiceMetadata;
```

### usage
```typescript
<ServiceMetadata
  servicesMetadata={metadata}
  getItemOnPress={handleItemOnPress}
  serviceId={service.service_id}
  organizationFiscalCode={service.organization_fiscal_code}
/>
```

Reply:
Makes sense, fixed as per our chat.